src/chttpd/src/chttpd_db.erl

```diff
@@ -1819,6 +1828,18 @@ db_attachment_req(#httpd{method = Method} = Req, Db, DocId, FileNameParts) when
         )
     ),
 
+    case Method of
+        'DELETE' ->
+            case get_attachment(Req, Db, DocId, FileName) of
```

Comment:
Looks much better, I think!

We can improve it a bit by moving it under the `case Method of 'DELETE' of ` clause since we already have a clause for DELETE and we're not doing anything but returning `[]` so instead we could check for existence there

Reply:
Moved in https://github.com/apache/couchdb/pull/3667/commits/222f1c21573ee3067ddc63df437f8a591d8c4300.